deepchecks/vision/checks/performance/performance_report.py

```diff
@@ -0,0 +1,259 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing performance report check."""
+from typing import Callable, TypeVar, List
+import pandas as pd
+import plotly.express as px
+from ignite.metrics import Metric
+
+from deepchecks.core import CheckResult, ConditionResult
+from deepchecks.vision import TrainTestBaseCheck
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.utils.strings import format_percent, format_number
+from deepchecks.vision import VisionDataset
+from deepchecks.vision.dataset import TaskType
+from deepchecks.vision.utils.metrics import get_scorers_list, task_type_check, calculate_metrics
+
+
+__all__ = ['PerformanceReport']
+
+from deepchecks.vision.utils.validation import model_type_validation
+
+PR = TypeVar('PR', bound='PerformanceReport')
+
+
+class PerformanceReport(TrainTestBaseCheck):
+    """Summarize given metrics on a dataset and model.
+
+    Args:
+        alternative_metrics (List[Metric], default None):
+            A list of ignite.Metric objects whose score should be used. If None are given, use the default metrics.
+    """
+
+    def __init__(self, alternative_metrics: List[Metric] = None, prediction_extract: Callable = None):
+        super().__init__()
+        self.alternative_metrics = alternative_metrics
+        self.prediction_extract = prediction_extract
+
+    def run(self, train_dataset: VisionDataset, test_dataset: VisionDataset, model=None) -> CheckResult:
+        """Run check.
+
+        Args:
+            train_dataset (VisionDataset): a VisionDataset object
+            test_dataset (VisionDataset): a VisionDataset object
+            model : A torch model supporting inference in eval mode.
+
+        Returns:
+            CheckResult: value is dictionary in format 'score-name': score-value
+        """
+        return self._performance_report(train_dataset, test_dataset, model)
+
+    def _performance_report(self, train_dataset: VisionDataset, test_dataset: VisionDataset, model):
+        VisionDataset.validate_dataset(train_dataset)
+        VisionDataset.validate_dataset(test_dataset)
+        train_dataset.validate_label()
+        test_dataset.validate_label()
+        train_dataset.validate_shared_label(test_dataset)
+        model_type_validation(model)
```

Comment:
TODO: add validation that self.prediction_extract(model(batch[0])) returns the format we want. Should be in metric utils or something.